sources/libs/cc/proc/macro.c

```diff
@@ -0,0 +1,163 @@
+#include <cc/parse.h>
+#include <cc/proc/macro.h>
+#include <cc/proc/proc.h>
+
+void cproc_parse_define(Lex *source, CProcContext *ctx)
+{
+    CProcMacro macro = {};
+    vec_init(&macro.args, ctx->alloc);
+    vec_init(&macro.source, ctx->alloc);
+
+    cparse_skip_separator(source, CLEX_DEFINE);
+
+    if (lex_curr(source).type != CLEX_IDENT)
```

Comment:
Any token can be redefined in C like: `#define + -` ;)

Reply:
oh crap